### DISCUSSION TITLE
How to mock composable method in a component?

### DISCUSSION THREAD

**Discussion by qashier-ray-foo** [vuejs/test-utils#2094] [Q&A]
I tried `vi.mock("@/composables/useSampleComposable")` but the composable is not mocked. Need help

**Comment by cexbrayat** [comment]
As you can see in `vi.mock` documentation, you need to add second parameter which is the factory of the mock object you want to use https://vitest.dev/api/vi.html#vi-mock

It's usually something like:
```ts
vi.mock("@/composables/useSampleComposable", () =>({
  mySuperComposableFunction: vi.fn()
}))
```

**Comment by qashier-ray-foo** [comment]
> As you can see in `vi.mock` documentation, you need to add second parameter which is the factory of the mock object you want to use https://vitest.dev/api/vi.html#vi-mock
> 
> It's usually something like:
> 
> ```ts
> vi.mock("@/composables/useSampleComposable", () =>({
>   mySuperComposableFunction: vi.fn()
> }))
> ```
I tried this method before, but when testing the component, the composable still is not mocked. 😅

**Reply by cexbrayat** [reply]
If you want help from us, you can build a repro online with https://stackblitz.com/github/vuejs/create-vue-templates/tree/main/typescript-vitest?file=src%2Fcomponents%2F__tests__%2FHelloWorld.spec.ts and we'll take a look

**Reply by qashier-ray-foo** [reply]
Could it be because I am using quasar webpack with vitest? I forked the repro and install all the quasar libs, using vite project with vitest, I am able to mock the composable
